src/core/control/xojfile/LoadHandler.cpp

```diff
@@ -1018,51 +1026,48 @@ auto LoadHandler::loadDocument(fs::path const& filepath) -> Document* {
     return &this->doc;
 }
 
-// Todo(fabian): return data and length by value not by reference, to ensure data and length is assigned always
-//      return string not a pointer. Ownage is not clear!
-auto LoadHandler::readZipAttachment(fs::path const& filename, gpointer& data, gsize& length) -> bool {
+auto LoadHandler::readZipAttachment(fs::path const& filename)
+    -> std::optional<std::pair<gpointer, gsize>> {
```

Comment:
I think the `gpointer` and `gsize` types should be removed from this function (that has nothing to do with gtk).
It should also return an `std::string` directly.